DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -178,34 +177,34 @@ function Set-TargetResource
 
             if (-not (Test-RuleProperties -FirewallRule $firewallRule @PSBoundParameters))
             {
-                # Effectively renaming DisplayGroup to Group
-                if ($DisplayGroup) {
-                    $null = $PSBoundParameters.Add('Group', $DisplayGroup)
-                }
-
                 Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
                     $($LocalizedData.UpdatingExistingFirewallMessage) -f $Name
                     ) -join '')
 
-                # If the DisplayGroup is being changed then the rule
-                # Has to be removed and recreated because that is the
-                # Only way the DisplayGroup can be changed unfortunately.
-                # A change to the set-netfirewallrule has been requested:
-                # https://connect.microsoft.com/PowerShell/feedbackdetail/view/1970765/add-ability-to-change-firewall-displaygroup-in-set-netfirewallrule-cmdlet
+                # If the Group is being changed the the rule needs to be recreated
                 if ($PSBoundParameters.ContainsKey('Group') `
                     -and ($Group -ne $FirewallRule.Group)) {
-                    # Although it is possible to automatically remove and recreate
-                    # this rule, it introduces more problems than it solves.
-                    # So for now this will throw an error with the suggested solution.
-                    $errorId = 'CantChangeDisplayGroupError'
-                    $errorCategory = [System.Management.Automation.ErrorCategory]::InvalidOperation
-                    $errorMessage = $($LocalizedData.CantChangeDisplayGroupError) -f $Name
-                    $exception = New-Object -TypeName System.InvalidOperationException `
-                        -ArgumentList $errorMessage
-                    $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
-                        -ArgumentList $exception, $errorId, $errorCategory, $null
-
-                    $PSCmdlet.ThrowTerminatingError($errorRecord)
+
+                    Remove-NetFirewallRule -Name $Name
+
+                    # Merge the existing rule values into the PSBoundParameters
+                    # so that it can be splatted.
+
+                    # Loop through each possible property and if it is not passed as a parameter
+                    # then set the PSBoundParameter property to the exiting rule value.
+                    'DisplayName','Group','Enabled','Action','Profile','Direction', `
```

Comment:
Er. Right, that's what I meant. :grinning:  Yes, let's get that fixed up, when we get this submitted let's get an issue open to fix it once we've added support for the other parameters (or do it as part of the support I guess :+1: )